pact/coin-contract/v4/coin-v4.pact

```diff
@@ -469,9 +469,7 @@
       (if (= "" r) true
```

Comment:
hmm ... the point is to stop parsing in pact altogether, really even the `check-reserved` call is slow. Suggest instead that we test `validate-principal` in a `let`, and if `true` return fast. if `false`, _then_ `check-reserved` and only fail if reserved (otherwise it's vanity so pass).

Reply:
Works for me - I figured we'd want to optimize for the non-protocol account branch, but if we want to optimize for protocols, that works.